src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -284,7 +284,22 @@ def _alloc_from_dict(self, config_dict):
                 ) from err
             if args:
                 job.private_args = SubstitutionList()
-                job.private_args.add_from_string(args)
+                error_message = job.private_args.add_from_string(args)
+                if error_message:
+                    defines = config_dict.get(ConfigKeys.DEFINE_KEY, [])
+                    conffile, confpath = "", ""
+
+                    for defs in defines:
+                        if defs[0] == "<CONFIG_FILE>":
+                            conffile = defs[1]
+                        elif defs[0] == "<CONFIG_PATH>":
+                            confpath = defs[1]
+
+                    raise ConfigValidationError(
+                        f"{error_message}: FORWARD_MODEL {job_name} with arguments "
+                        f"{args} originating from file {confpath}/{conffile}"
```

Comment:
Error message should handle the case where `<CONFIG_FILE>` or `<CONFIG_FILE>` is not in defines gracefully. Also, instead of looking for it in the config_dict, it should be possible to get it from `self.substitution_list`.

Reply:
Updated. :+1: